snuba/cli/subscriptions.py

```diff
@@ -0,0 +1,195 @@
+
+
+@click.command()
+@click.option(
+    "--dataset",
+    "dataset_name",
+    default="events",
+    type=click.Choice(DATASET_NAMES),
+    help="The dataset to target",
+)
+@click.option("--topic")
+@click.option("--partitions", type=int)
+@click.option("--commit-log-topic")
+@click.option(
+    "--commit-log-group",
+    "commit_log_groups",
+    multiple=True,
+    default=["snuba-consumers"],
+)
+@click.option(
+    "--consumer-group",
+    default="snuba-subscriptions",
+    help="Consumer group use for consuming the dataset source topic.",
+)
+@click.option(
+    "--auto-offset-reset",
+    default="error",
+    type=click.Choice(["error", "earliest", "latest"]),
+    help="Kafka consumer auto offset reset.",
+)
+@click.option(
+    "--bootstrap-server",
+    "bootstrap_servers",
+    multiple=True,
+    help="Kafka bootstrap server(s) to use.",
+)
+@click.option(
+    "--max-batch-size",
+    default=settings.DEFAULT_MAX_BATCH_SIZE,
+    type=int,
+    help="Max number of messages to batch in memory before writing to Kafka.",
+)
+@click.option(
+    "--max-batch-time-ms",
+    default=settings.DEFAULT_MAX_BATCH_TIME_MS,
+    type=int,
+    help="Max length of time to buffer messages in memory before writing to Kafka.",
+)
+@click.option("--schedule-ttl", type=int, default=60 * 5)
+@click.option("--result-topic")
+@click.option("--log-level", default=settings.LOG_LEVEL, help="Logging level to use.")
+def subscriptions(
+    *,
+    dataset_name: str,
+    topic: Optional[str],
+    partitions: Optional[int],
+    commit_log_topic: Optional[str],
+    commit_log_groups: Sequence[str],
+    consumer_group: str,
+    auto_offset_reset: str,
+    bootstrap_servers: Sequence[str],
+    max_batch_size: int,
+    max_batch_time_ms: int,
+    schedule_ttl: int,
+    result_topic: Optional[str],
+    log_level: str,
+) -> None:
+    assert result_topic is not None
+
+    setup_logging(log_level)
+    setup_sentry()
+
+    dataset = get_dataset(dataset_name)
+
+    if not bootstrap_servers:
+        bootstrap_servers = settings.DEFAULT_DATASET_BROKERS.get(
+            dataset_name, settings.DEFAULT_BROKERS
+        )
+
+    producer = KafkaProducer(
+        {
+            "bootstrap.servers": ",".join(bootstrap_servers),
+            "partitioner": "consistent",
+            "message.max.bytes": 50000000,  # 50MB, default is 1MB
+        },
+        SubscriptionResultCodec(),
+    )
+
+    loader = enforce_table_writer(dataset).get_stream_loader()
+
+    with closing(producer):
+        consumer = BatchingConsumer(
+            TickConsumer(
+                SynchronizedConsumer(
+                    KafkaConsumer(
+                        build_kafka_consumer_configuration(
+                            bootstrap_servers,
+                            consumer_group,
+                            auto_offset_reset=auto_offset_reset,
+                        ),
+                        PassthroughCodec(),
```

Comment:
This doesn't need to do any decoding of the message, so this becomes `Consumer[KafkaPayload]`, which meets the `TickConsumer`'s type constraint of `Consumer[Any]`: https://github.com/getsentry/snuba/blob/master/snuba/subscriptions/consumer.py#L76